src/Breakdown/MetricLabelValuesList/SceneByFrameRepeater.tsx

```diff
@@ -0,0 +1,284 @@
+      counts: { current: 0, total: 0 },
+    });
+
+    this.addActivationHandler(() => {
+      const dataProvider = sceneGraph.getData(this);
+      if (!dataProvider) {
+        throw new Error('No data provider found!');
+      }
+
+      this.initFilterAndSort();
+
+      this._subs.add(
+        dataProvider.subscribeToState((newState) => {
+          if (newState.data) {
+            this.performRepeat(newState.data);
+          }
+        })
+      );
+
+      if (dataProvider.state.data) {
+        this.performRepeat(dataProvider.state.data);
+      }
+    });
+  }
+
+  private performRepeat(data: PanelData) {
+    if (data.state === LoadingState.Loading) {
+      this.setState({
+        loadingLayout: this.state.getLayoutLoading?.(),
+        errorLayout: undefined,
+        emptyLayout: undefined,
+        currentBatchSize: 0,
+      });
+      return;
+    }
+
+    if (data.state === LoadingState.Error) {
+      this.setState({
+        errorLayout: this.state.getLayoutError?.(data),
+        loadingLayout: undefined,
+        emptyLayout: undefined,
+        currentBatchSize: 0,
+      });
+      return;
+    }
+
+    const filteredSeries = this.filterAndSort(data.series);
+
+    if (!filteredSeries.length) {
+      this.setState({
+        emptyLayout: this.state.getLayoutEmpty?.(),
+        errorLayout: undefined,
+        loadingLayout: undefined,
+        currentBatchSize: 0,
+        counts: { current: 0, total: data.series.length },
+      });
+      return;
+    }
+
+    this.setState({
+      loadingLayout: undefined,
+      errorLayout: undefined,
+      emptyLayout: undefined,
+      currentBatchSize: this.state.initialPageSize,
+      counts: { current: filteredSeries.length, total: data.series.length },
+    });
+
+    const newChildren: SceneObject[] = filteredSeries
+      .slice(0, this.state.initialPageSize)
+      .map((s, i) => this.state.getLayoutChild(data, s, i))
+      .filter(Boolean) as SceneObject[];
+
+    this.state.body.setState({ children: newChildren });
+  }
+
+  private initFilterAndSort() {
+    this.searchText = sceneGraph.findByKeyAndType(this, 'quick-search', QuickSearch).state.value;
+    this.sortBy = sceneGraph.findByKeyAndType(this, 'breakdown-sort-by', SortBySelector).state.value.value;
+  }
+
+  private filterAndSort(series: PanelData['series']) {
+    let filteredSeries: DataFrame[] = [];
+
+    if (!this.searchText) {
+      filteredSeries = series;
+    } else {
+      const regexes = this.searchText
+        .split(',')
+        .map((p) => p.trim())
+        .filter(Boolean)
+        .map((r) => {
+          try {
+            return new RegExp(r);
+          } catch {
+            return null;
+          }
+        })
+        .filter(Boolean) as RegExp[];
+
+      for (let i = 0; i < series.length; i += 1) {
+        const s = series[i];
+
+        if (regexes.some((regex) => regex.test(getLabelValueFromDataFrame(s)))) {
+          filteredSeries.push(s);
+        }
+      }
```

Comment:
Seems like this could be boiled down to:

```suggestion
      filteredSeries = series.filter((frame) => regexes.some((regex) => regex.test(getLabelValueFromDataFrame(frame))));
```

Reply:
Indeed. I actually left it as it is after experimenting with storing the series index L189 in order to preserve the series colors before/after filtering. There was some edge cases though... I'd like to come back to the idea in the future.